File: src/main/java/com/dongfg/project/api/config/JobInitConfig.java
package com.dongfg.project.api.config;

import com.dongfg.project.api.job.BaseScheduleJob;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.event.EventListener;
import org.springframework.util.CollectionUtils;

import java.util.List;

/**
 * @author dongfg
 * @date 18-1-3
 */
@Configuration
public class JobInitConfig {
    @Autowired(required = false)
    private List<BaseScheduleJob> jobs;

    @EventListener(ApplicationReadyEvent.class)
    public void initJob() {
        if (CollectionUtils.isEmpty(jobs)) {
            return;
        }
        jobs.forEach(BaseScheduleJob::submitJob);
    }
}

File: src/main/kotlin/com/dongfg/project/api/common/config/JobInitConfig.kt
package com.dongfg.project.api.common.config

import com.dongfg.project.api.service.job.BaseJob
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.annotation.Configuration
import org.springframework.context.event.EventListener

@Configuration
class JobInitConfig {
    @Autowired(required = false)
    private lateinit var jobs: List<BaseJob>

    @EventListener(ApplicationReadyEvent::class)
    fun initJob() {
        jobs.forEach(BaseJob::submit)
    }
}